torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1241,6 +1241,17 @@ def _advise_is_size(a: SymInt) -> None:
         _constrain_range_for_size(a)
 
 
+def _advise_is_bounded(a: SymInt, upper_bound: Union[int, SymInt]) -> None:
+    if (
+        isinstance(a, SymInt)
```

Comment:
don't we (shouldn't we) have a util that directly checks whether something is a symbol? also weird that type annotation says it is a symint and we still check it here

Reply:
Yeah, probably would be fine to have a util along those lines. Re the type annotation, just because we are type checking, doesn't mean the user is type checking!

Reply:
I attempted the util but it is inconvenient as the explicit isinstance checks do type narrowing which you would lose from the utility.